asset.go

```diff
@@ -104,3 +114,8 @@ func (a *Asset) Sink() pipe.SinkAllocatorFunc {
 		}, nil
 	}
 }
+
+// Source is a convenience wrapper to audio.Source
+func (a *Asset) Source() pipe.SourceAllocatorFunc {
```

Comment:
I understand your point about consistency with other components, but I came up to conclusion that most of the time `Source`, `Sink` and `Processor` methods don't really need to have a receiver. For example: [wav package](https://github.com/pipelined/wav/pull/10/files). Sink and Source structs were just holding references to Reader/Writer which can be easily replaced with functions parameters. This design makes it very clear how `wav.Source` and `wav.Sink` should be used: [only to construct](https://github.com/pipelined/wav/pull/10/files#diff-38a154a3d086ffc7fb1b89cf22b4af05R51-R52) `pipe.Routing`.

With the `audio.Asset` things are a bit more tricky, because in contrast to `wav.Source` or `wav.Sink` it has much more complicated lifecycle - user can sink signal into it, use it to compose tracks or source the signal directly from it. That's why I decided to have only `asset.Sink` method - it will make the lifecycle of the asset very explicit. 

Please, let me know if you have other thoughts about this.

Reply:
Thinking about it from a implementation in a 3rd party application experience having a consistent interface across the lib gives a much better developer experience.

I see your point, however there's 2 distinct differences between asset and wav as examples. wav.Source can only be used wiht the source allocator function since that handles decoding at the point it's it accessed.

Asset on the other hand is a container for signal buffers so perhaps source and sink in themselves are what's not clear here?

Reply:
Can you please clarify what do you exactly mean with Source/Sink? 

I'll resolve this conversation, because I think that we have two threads about basically the same things.